nsq/protocol.py

```diff
@@ -70,15 +71,24 @@ def decode_message(data):
 
 
 def _command(cmd, body, *params):
-    body_data = ''
-    params_data = ''
+    body_prefix = b''
+    params_data = b''
     if body:
-        assert isinstance(body, str), 'body must be a string'
-        body_data = struct.pack('>l', len(body)) + body
+        assert isinstance(body, compat.string_like), 'body must be a string'
+        if isinstance(body, compat.unicode):
+            body = body.encode('utf-8')
```

Comment:
Yeah, I saw. I instead documented the casting in `Writer.pub` and `Writer.mpub` (there may be other places, as well?), which was mentioned in the previous discussion. 

But I can require bytes if you like, though I would tend to think that this will handle the most common case conveniently. If you have unicode or bytes, as you almost certainly do, it works fine; if you don't, read the docs and pass in bytes.